eif_defs/src/lib.rs

```diff
@@ -26,9 +26,10 @@ pub const EIF_HDR_ARCH_ARM64: u16 = 0x1;
 
 /// Current EIF version to be incremented every time we change the format
 /// of this structures, we assume changes are backwards compatible.
-/// V0 -> V1: Add support to generate and check CRC.
-/// V1 -> V2: Add the signature section.
-pub const CURRENT_VERSION: u16 = 3;
+/// V1 -> V2: Add support to generate and check CRC.
+/// V2 -> V3: Add the signature section.
+/// V3 -> V4: Add the metadata section.
+pub const CURRENT_VERSION: u16 = 4;
```

Comment:
Does the Enclave device expose the supported EIF version? Should nitro-cli try to make images compatible with older versions if the new features aren't required?

Reply:
No, looks like there is no any exposal of supported EIF versions yet.
Specifically for metadata section VMM checks internally the version of loaded eif and extracts the metadata only if the version >= 4. As an alternative we could make metadata section fully optional even for new EIF version.
But, indeed, better to think in the future about some compatibility check mechanism.